docs/explanation/e-hardening.md

```diff
@@ -0,0 +1,119 @@
+| AWS       | [Juju AWS Permission](https://discourse.charmhub.io/t/juju-aws-permissions/5307), [AWS Instance Profiles](https://discourse.charmhub.io/t/using-aws-instance-profiles-with-juju-2-9/5185), [Juju on AWS](https://juju.is/docs/juju/amazon-ec2) | 
+| Azure     | [Juju Azure Permission](https://juju.is/docs/juju/microsoft-azure), [How to use Juju with Microsoft Azure](https://discourse.charmhub.io/t/how-to-use-juju-with-microsoft-azure/15219)                                                                                                  |
+
+#### Juju users
+
+It is very important that the different juju users are set up with minimal permission depending on the scope for their operations. 
+Please refer to the [User access levels](https://juju.is/docs/juju/user-permissions) documentation for more information on the access level and corresponding abilities 
+that the different users can be granted. 
+
+Juju user credentials must be stored securely and rotated regularly to prevent leakage and unauthorized access. 
+
+## Applications
+
+In the following we provide guidance on how to harden your deployment using:
+
+1. Operating System
+2. Kafka and ZooKeeper Security Upgrades
+3. Encryption 
+4. Authentication
+5. Monitoring and Auditing
+
+### Operating System
+
+Charmed Kafka and Charmed ZooKeeper currently run on top of Ubuntu 22.04. Deploy a [Landscape Client Charm]() in order to 
+connect the underlying VM to a Landscape User Account to manage security upgrades and integrate Ubuntu Pro subscriptions. 
+
+### Kafka and ZooKeeper Security Upgrades
+
+Charmed Kafka and Charmed ZooKeeper operators install a pinned revision of the [Charmed Kafka snap](https://snapcraft.io/charmed-kafka)
+and [Charmed ZooKeeper snap](https://snapcraft.io/charmed-zookeeper), respectively, in order to provide reproducible and secure environments. 
+New versions of Charmed Kafka and Charmed ZooKeeper may be released to provide patching of vulnerabilities (CVEs). 
+It is important to refresh the charm regularly to make sure the workload is as secure as possible. 
+For more information on how to refresh the charm, please refer to the [how-to upgrade](https://charmhub.io/kafka/docs/h-upgrade) documentation.
+
+### Encryption
+
+Charmed Kafka must be deployed with security enabled. This requires to deploy Kafka and ZooKeeper and relate them to one of the TLS certificate operator charms. 
+Please refer to the [Charming Security page](https://charmhub.io/topics/security-with-x-509-certificates) for more information on how to select the right certificate
+provider for your use-case. 
+
+For more information, please refer to the [how to enable encryption](https://charmhub.io/kafka/docs/h-enable-encryption) user guide.
+
+### Authentication
+
+Charmed Kafka supports the following authentication layers:
+
+1. [SCRAM-based SASL Authentication](/t/charmed-kafka-how-to-manage-app/10285)
```

Comment:
```suggestion
1. [SCRAM-based SASL Authentication](/t/10285)
```
This is the recommended way to do the links to the Discourse. It's shorter and easier to use, but lacks context of the link in the URL, but it makes it more resilient to errors and changes. It is not mandatory, but it is recommended.

Feel free to ignore or apply to all similar links in the file.

Reply:
Really? I didn't know that was the recommendation. Do you have a link for this?

If there is not a strong guidance, I would honestly prefer to have the slug which provides a bit more information than an ID only.

Reply:
That's not a requirement, so feel free to use the full version with the slug. 
I don't have any written evidence ready at the moment, I just recall a discussion we had with other TAs on this. The main reason for the shorter version being the recommended one, I believe, was that it looks much simpler (encourages people to contribute) and less prone to errors.